### PR TITLE
Change way of doing cicd

### PR DESCRIPTION
**Pull request by valentinmadrid**

### CODE REVIEW

**Comment by valentinmadrid** [comment]
A good way of doing checks is finally done. These github actions can be ran once a month and build every program in this repository. Cost wise, I think we are staying below the GitHub actions free tier 1000 minutes per month, which is good.
Here is how it works:
- For every action(every framework has a different action), the necessary dependencies get installed.
- Then a loop gets created to build each program one after the other.
- If there is any issue, the build fails and the maintainers get notified

**Comment by valentinmadrid** [comment]
To-do now is fixing all of the issues that are encountered when running these workflows: https://github.com/solana-developers/program-examples/pull/34

File: .github/workflows/anchor.yml [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
+          sh -c "$(curl -sSfL https://release.solana.com/stable/install)"
+          export PATH="/home/runner/.local/share/solana/install/active_release/bin:$PATH"
+          solana -V
+          rustc -V
+          sudo apt-get update && sudo apt-get upgrade && sudo apt-get install -y pkg-config build-essential libudev-dev libssl-dev
+          cargo install --git https://github.com/coral-xyz/anchor avm --locked --force
+          avm install latest
+          avm use latest
+          declare -a StringArray=(
+            #"basics/checking-accounts/anchor"
```
Comment (buffalojoec) [inline .github/workflows/anchor.yml:26]: Why are the hash marks necessary `#`?

File: .github/workflows/native.yml [hunk 2]
```diff
@@ -1,60 +1,53 @@
...
-        working-directory: ${{ matrix.dir }}
-        run: cargo build-sbf --verbose
+      - uses: actions/checkout@v2
+      
+      - name: Install dependencies
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/beta/install)"
+          export PATH="/home/runner/.local/share/solana/install/active_release/bin:$PATH"
+          solana -V
+          rustc -V
```
Comment (buffalojoec) [inline .github/workflows/native.yml:20]: How is Rust installed here already?

File: .github/workflows/native.yml [hunk 3]
```diff
@@ -1,60 +1,53 @@
...
-        uses: actions/checkout@v3
-
-      - name: Build program
-        working-directory: ${{ matrix.dir }}
-        run: cargo build-sbf --verbose
+      - uses: actions/checkout@v2
+      
+      - name: Install dependencies
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/beta/install)"
```
Comment (buffalojoec) [inline .github/workflows/native.yml:17]: Beta version or latest? Which would make more sense for the examples?

File: .github/workflows/anchor.yml [hunk 4]
```diff
@@ -1,53 +1,53 @@
...
+            #"basics/counter/anchor"
+            #"basics/create-account/anchor"
+            #"basics/hello-solana/anchor"
+            #"basics/pda-rent-payer/anchor"
+            #"basics/processing-instructions/anchor"
+            #"basics/program-derived-addresses/anchor"
+            #"basics/rent/anchor"
+            #"basics/repository-layout/anchor"
+            #"basics/transfer-sol/anchor"
+            #!!!!"compression/cnft-vault/anchor"
```
[the remark this answers is not in the document]
Comment (valentinmadrid) [reply .github/workflows/anchor.yml:37]: You came at the wrong moment lol, I was doing a workaround as my fixes aren't merged yet. I'll let you know again whenever you can merge. Also drafting pr

[the remark this answers is not in the document]
Comment (buffalojoec) [reply .github/workflows/anchor.yml:37]: OK it said you requested my review

**Pull request merged**